module/zfs/vdev_queue.c

```diff
@@ -706,12 +698,41 @@ vdev_queue_aggregate(vdev_queue_t *vq, zio_t *zio)
 	aio->io_timestamp = first->io_timestamp;
 
 	nio = first;
+	next_offset = first->io_offset;
 	do {
 		dio = nio;
 		nio = AVL_NEXT(t, dio);
 		zio_add_child(dio, aio);
 		vdev_queue_io_remove(vq, dio);
+
+		if (dio->io_offset != next_offset) {
```

Comment:
Can you add a small block comment above this function summarizing what's going on in this loop and how we're assembling the multi-abd.